### ISSUE TITLE
Promise based api

### ISSUE THREAD

**Issue by iapain** [fagbokforlaget/pdftotextjs#9]
We should refactor this library to have cleaner promise based API. e.g.

``` javascript
var pdftotext = require('pdftotextjs');
var converter = new pdftotextjs('test/pdfs/sample.pdf');

converter.getText().then(function() {
  console.log("Success");
}).catch(function(err) {
  console.error("Conversion error: " + err);
});

```

@lirantal What you think about it?

**Comment by lirantal** [comment]
Awesome, I like it!

**Comment by iapain** [comment]
Implemented in v0.5.0
